syne_tune/experiments/plotting.py

```diff
@@ -0,0 +1,664 @@
+            new_params["subplots"] = default_params.subplots
+        elif default_params.subplots is None:
+            new_params["subplots"] = self.subplots
+        else:
+            new_params["subplots"] = self.subplots.merge_defaults(
+                default_params.subplots
+            )
+        if self.show_init_trials is None:
+            new_params["show_init_trials"] = default_params.show_init_trials
+        elif default_params.show_init_trials is None:
+            new_params["show_init_trials"] = self.show_init_trials
+        else:
+            new_params["show_init_trials"] = self.show_init_trials.merge_defaults(
+                default_params.show_init_trials
+            )
+        return PlotParameters(**new_params)
+
+
+class ComparativeResults:
+    """
+    This class loads, processes, and plots results of a comparative study,
+    combining several experiments for different methods, seeds, and
+    benchmarks (optional). Note that an experiment corresponds to one run
+    of HPO, resulting in files :const:`~syne_tune.constants.ST_METADATA_FILENAME`
+    for metadata, and :const:`~syne_tune.constants.ST_RESULTS_DATAFRAME_FILENAME`
+    for time-stamped results.
+
+    There is one comparative plot per benchmark (aggregation of results
+    across benchmarks are not supported here). Results are grouped by
+    setup (which usually equates to method), and then summary statistics are
+    shown for each setup as function of wall-clock time. The plot can also
+    have several subplots, in which case results are first grouped into
+    subplot number, then setup.
+
+    If ``benchmark_key is None``, there is only a single benchmark, and all
+    results are merged together.
+
+    Both setup name and subplot number (optional) can be configured by the
+    user, as function of metadata written for each experiment. The functions
+    ``metadata_to_setup`` and ``metadata_to_subplot`` (optional) can also be
+    used for filtering: results of experiments for which any of them returns
+    ``None``, are not used.
+
+    When grouping results w.r.t. benchmark name and setup name, we should end
+    up with ``num_runs`` experiments. These are (typically) random repetitions
+    with different seeds. If after grouping, a different number of experiments
+    is found for some setup, a warning message is printed. In this case, we
+    recommend to check the completeness of result files. Common reasons:
+
+    * Less than ``num_runs`` experiments found. Experiments failed, or files
+      were not properly synced.
+    * More than ``num_runs`` experiments found. This happens if initial
+      experiments for the study failed, but ended up writing results. This can
+      be fixed by either removing the result files, or by using
+      ``datetime_bounds`` (since initial failed experiments ran first).
+
+    Result files have the path
+    ``f"{experiment_path()}{ename}/{patt}/{ename}-*/"``, where ``path`` is from
```

Comment:
This assumes that all runs of one method are in some subfolder and the runs of another method in another subfolder, correct?

Reply:
Depends a bit. If a method is cheap to run (e.g., ASHA in simulation), then results land in the same folder for all seeds. For expensive ones, you have separate subfolders per seed.